src/hip_module.cpp

```diff
@@ -149,6 +149,15 @@ hipError_t ihipModuleLaunchKernel(TlsData *tls, hipFunction_t f, uint32_t global
         ret = hipErrorInvalidDevice;
 
     } else {
+
+        // Kernel launch may fail for (no. of VGPRs X blocksize >= 65536)
+        if(f != nullptr && f->_header != nullptr){
+            uint32_t usedVGPRCount = f->_header->workitem_vgpr_count;
+            if((localWorkSizeX * localWorkSizeY * localWorkSizeZ * usedVGPRCount) >= 65536){
```

Comment:
Please avoid hardcoded constants in code, instead use const MAX_VGPR_SIZE or the like with description of the reasoning behind that magic value.